Fix a couple of heap overflow vulnerabilities, all "disputed" severity
https://github.com/godotengine/godot/issues/123608
CVE-2026-110734
CVE-2026-110735
CVE-2026-110736

Index: modules/hdr/image_loader_hdr.cpp
--- modules/hdr/image_loader_hdr.cpp.orig
+++ modules/hdr/image_loader_hdr.cpp
@@ -60,6 +60,10 @@ Error ImageLoaderHDR::load_image(Ref<Image> p_image, R
 
 	int width = f->get_line().to_int();
 
+	ERR_FAIL_COND_V_MSG(width <= 0 || height <= 0, ERR_FILE_CORRUPT, "Invalid image size, corrupt HDR.");
+	ERR_FAIL_COND_V_MSG(width > Image::MAX_WIDTH || height > Image::MAX_HEIGHT || (int64_t)width * height > Image::MAX_PIXELS, ERR_UNAVAILABLE,
+			vformat("The HDR image size (%dx%d) exceeds the maximum image size.", width, height));
+
 	Vector<uint8_t> imgdata;
 
 	imgdata.resize(height * width * (int)sizeof(uint32_t));
@@ -107,11 +111,13 @@ Error ImageLoaderHDR::load_image(Ref<Image> p_image, R
 							// Run
 							int value = f->get_8();
 							count -= 128;
+							ERR_FAIL_COND_V_MSG(count > width - i, ERR_FILE_CORRUPT, "Invalid run length, corrupt HDR.");
 							for (int z = 0; z < count; ++z) {
 								ptr[(j * width + i++) * 4 + k] = uint8_t(value);
 							}
 						} else {
 							// Dump
+							ERR_FAIL_COND_V_MSG(count == 0 || count > width - i, ERR_FILE_CORRUPT, "Invalid dump length, corrupt HDR.");
 							f->get_buffer(temp_read_ptr, count);
 							for (int z = 0; z < count; ++z) {
 								ptr[(j * width + i++) * 4 + k] = temp_read_ptr[z];
